cookbook/form/create_form_type_extension.rst

```diff
@@ -319,3 +311,15 @@ next to the file field. For example::
 
 When displaying the form, if the underlying model has already been associated
 with an image, you will see it displayed next to the file input.
+
+Generic Form Type Extensions
+----------------------------
+
+Although it is not possible to have a form type extension applying to all form
+types, **most** form types natively available in Symfony
+(:doc:`/reference/forms/types`) inherit from the ``form`` form type. Thus, a
+form type extension applying to ``form`` would apply to all of these.
+
+A notable exception are the ``button`` from types. Plus, keep in mind that a custom
+form type which doesn't inherit neither ``form`` nor ``button`` could always be
+created.
```

Comment:
Unless I'm missing some other spot, I think this is actually our opportunity to introduce the "feature" where you can apply extensions to most field types, instead of telling them what they *can't* do (that would make sense if the rest of the article had been an example using the `form` extension, and here we were warning them of the edge cases).

So, I'd like to see this paragraph introduce the feature first, and then the warning, something like:

>
You can modify all (well, most - see below) field types with a form type extension by making `getExtendedType()` return `form`: the base type for *most* Symfony field types. But, be aware that it *is* possible for a field type to *not* extend `form`, and thus for your extension to not apply to *all* fields.

... then the "A notable exception..."

What do you think?